### PR TITLE
CVSL-1365 Adding arch tests

### PR DESCRIPTION
**Pull request by andrewrlee**
Adding a couple of tests to ensure that roles have the correct role annotations and are tagged appropriately. 

Also removing untagged and deprecated statistic controller

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt [hunk 1]
```diff
@@ -25,7 +25,7 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.OffenderSer
 @RestController
 @RequestMapping("/offender", produces = [MediaType.APPLICATION_JSON_VALUE])
 class OffenderController(private val offenderService: OffenderService, private val comService: ComService) {
-  @Tag(name = Tags.COM)
+  @Tag(name = Tags.OFFENDER)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt:28]: Improving tags

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt [hunk 2]
```diff
@@ -110,6 +110,7 @@ class OffenderController(private val offenderService: OffenderService, private v
     this.offenderService.updateProbationTeam(crn, body)
   }
 
+  @Tag(name = Tags.OFFENDER)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt:113]: Adding missing tag

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/ControllerTagTest.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import io.swagger.v3.oas.annotations.tags.Tag
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.fail
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider
+import org.springframework.core.type.filter.AnnotationTypeFilter
+import org.springframework.web.bind.annotation.RestController
+import java.lang.reflect.AnnotatedElement
+import java.lang.reflect.Method
+
+class ControllerTagTest {
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/ControllerTagTest.kt:12]: A test that will fail if we add any new endpoints but forget to tag the controller or endpoint

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt [hunk 4]
```diff
@@ -23,9 +23,9 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.ComService
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.OffenderService
 
 @RestController
+@Tag(name = Tags.OFFENDER)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt:26]: All endpoints in this controller inherit the Offender Tag

This effectively adds a tag to the missing endpoint

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/EndpointSecurityTest.kt [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.fail
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider
+import org.springframework.core.type.filter.AnnotationTypeFilter
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.RestController
+import java.lang.reflect.AnnotatedElement
+import java.lang.reflect.Method
+
+class EndpointSecurityTest {
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/EndpointSecurityTest.kt:12]: A test that will fail if we add an endpoint and forget to add the appropriate role checks or [excluding annotation](https://github.com/ministryofjustice/create-and-vary-a-licence-api/pull/308/files#diff-70a75ff545f9111231777febe63f1dfbfcacd31f91510d73b5a328fade47bf0f)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceStatisticsController.kt
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceStatisticsController.kt:1]: Deleting unused controller

**Review by justin-gardiner — APPROVED** [review]
lgtm

**Pull request merged**